src/Simplify.elm

```diff
@@ -2876,6 +2911,154 @@ stringReverseChecks ({ parentRange, fnRange, firstArg } as checkInfo) =
+
+                     else
+                        Nothing
+                    )
+                |> Maybe.withDefault []
+
+        ( _, Nothing, _ ) ->
+            []
+
+
+stringLeftChecks : CheckInfo -> List (Error {})
+stringLeftChecks checkInfo =
+    case ( checkInfo.firstArg, checkInfo.secondArg ) of
+        ( _, Just (Node _ (Expression.Literal "")) ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.left on an empty string will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "\"\"" ]
+            ]
+
+        ( Node _ (Expression.Integer 0), _ ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.left with length 0 will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "always \"\"" ]
+            ]
+
+        ( Node _ (Expression.Negation (Node _ (Expression.Integer _))), _ ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.left with negative length will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "always \"\"" ]
+            ]
+
+        _ ->
+            []
+
+
+stringRightChecks : CheckInfo -> List (Error {})
+stringRightChecks checkInfo =
+    case ( checkInfo.firstArg, checkInfo.secondArg ) of
+        ( _, Just (Node _ (Expression.Literal "")) ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.right on an empty string will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "\"\"" ]
+            ]
+
+        ( Node _ (Expression.Integer 0), _ ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.right with length 0 will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "always \"\"" ]
+            ]
+
+        ( Node _ (Expression.Negation (Node _ (Expression.Integer _))), _ ) ->
+            [ Rule.errorWithFix
+                { message = "Using String.right with negative length will result in an empty string"
+                , details = [ "You can replace this call by an empty string." ]
+                }
+                checkInfo.fnRange
+                [ Fix.replaceRangeBy checkInfo.parentRange "always \"\"" ]
```

Comment:
This should probably be `""` if there is a second argument, and ``always ""`` if there isn't.